Track Your Reps/Models/User.swift

```diff
@@ -73,12 +75,49 @@ extension User {
     }
         
     fileprivate func resetEvents(for member: CongressMember, _ mergedEvents: [(Int, [CongressMember])]) -> CongressMember {
-        let newMember = member
-        newMember.events = newMember.events.map { event in
-            var newEvent = event
-            mergedEvents.forEach { if event.hashValue == $0.0 { newEvent.congressMembers = $0.1 } }
-            return newEvent
+        let memberCopy = member
+        memberCopy.events = memberCopy.events.map { event in
+            var eventCopy = event
+            mergedEvents.forEach { if event.hashValue == $0.0 { eventCopy.congressMembers = $0.1 } }
+            return eventCopy
+        }
+        return memberCopy
+    }
+    
+    fileprivate func fetchBills() {
+        let backgroundQueue = DispatchQueue.global()
+        let group = DispatchGroup()
+        
+        let eventSet = Set(dataStore.members.flatMap({ $0.events }))
+        var withBills = [Event]()
+        
+        for event in eventSet where event.isBill {
+            group.enter()
+            backgroundQueue.async(group: group, execute: {
+                event.fetchBill() { bill in
+                    var eventCopy = event
+                    eventCopy.bill = bill
+                    withBills.append(eventCopy)
+                    group.leave()
+                }
+            })
+        }
+        
+        group.notify(queue: DispatchQueue.main, execute: {
+            self.dataStore.members = self.dataStore.members.map { member in
+                return self.resetEvents(for: member, eventsWithBills: withBills)
+            }
+            self.delegate?.setupViewControllers()
+        })
+    }
+    
+    fileprivate func resetEvents(for member: CongressMember, eventsWithBills withBills: [Event]) -> CongressMember {
```

Comment:
reminder to myself: think I can shorten these resetEvents methods with a tip from ProSwift